tensorflow_addons/metrics/f_test.py

```diff
@@ -0,0 +1,128 @@
+
+        fbeta = FBetaScore(3, avg, beta, threshold)
+        self.evaluate(tf.compat.v1.variables_initializer(fbeta.variables))
+        self.evaluate(fbeta.update_state(act, pred))
+        return self.evaluate(fbeta.result())
+
+    def _test_sk(self, avg, beta, act, pred, threshold):
+        act = np.array(act)
+        pred = np.array(pred)
+        if threshold is None:
+            threshold = np.max(pred, axis=-1, keepdims=True)
+            pred = np.logical_and(pred >= threshold,
+                                  pred - 0 > 1e-12).astype('int')
+        else:
+            pred = (pred >= threshold).astype('int')
+
+        res = fbeta_score(act, pred, beta, average=avg)
+        return res
+
+    def _test_fbeta_score(self, actuals, preds, threshold=None):
+        for avg in [None, 'micro', 'macro', 'weighted']:
+            for beta_val in [0.5, 1.0, 2.0]:
+                tf_score = self._test_tf(avg, beta_val, actuals, preds,
+                                         threshold)
+                sk_score = self._test_sk(avg, beta_val, actuals, preds,
+                                         threshold)
+                self.assertAllClose(tf_score, sk_score, atol=1e-5)
+
+    def test_fbeta_perfect_score(self):
+        preds = [[0.7, 0.7, 0.7], [1, 0, 0], [0.9, 0.8, 0]]
+        actuals = [[1, 1, 1], [1, 0, 0], [1, 1, 0]]
+        self._test_fbeta_score(actuals, preds, 0.66)
+
+    def test_fbeta_worst_score(self):
+        preds = [[0.7, 0.7, 0.7], [1, 0, 0], [0.9, 0.8, 0]]
+        actuals = [[0, 0, 0], [0, 1, 0], [0, 0, 1]]
+        self._test_fbeta_score(actuals, preds, 0.66)
+
+    def test_fbeta_random_score(self):
+        preds = [[0.7, 0.7, 0.7], [1, 0, 0], [0.9, 0.8, 0]]
+        actuals = [[0, 0, 1], [1, 1, 0], [1, 1, 1]]
+        self._test_fbeta_score(actuals, preds, 0.66)
+
+    def test_fbeta_random_score_none(self):
+        preds = [[0.9, 0.1, 0], [0.2, 0.6, 0.2], [0, 0, 1], [0.4, 0.3, 0.3],
+                 [0, 0.9, 0.1], [0, 0, 1]]
+        actuals = [[1, 0, 0], [0, 1, 0], [0, 0, 1], [1, 0, 0], [1, 0, 0],
+                   [0, 0, 1]]
+        self._test_fbeta_score(actuals, preds, None)
+
+    def test_keras_model(self):
+        fbeta = FBetaScore(5, 'micro', 1.0)
+        utils.test_keras_model(fbeta, 5)
+
+
+@test_utils.run_all_in_graph_and_eager_modes
+class F1ScoreTest(tf.test.TestCase):
+    def test_eq(self):
+        f1 = F1Score(4)
+        fbeta = FBetaScore(4, beta=1.0)
+        self.evaluate(tf.compat.v1.variables_initializer(f1.variables))
+        self.evaluate(tf.compat.v1.variables_initializer(fbeta.variables))
+
+        actuals = np.random.randint(2, size=(10, 4))
+        preds = np.random.uniform(size=(10, 4))
+
+        self.evaluate(fbeta.update_state(actuals, preds))
+        self.evaluate(f1.update_state(actuals, preds))
+        self.assertAllClose(
+            self.evaluate(fbeta.result()), self.evaluate(f1.result()))
+
+    def test_keras_model(self):
```

Comment:
It seems that _get_model is designed to test model.evaluate: https://github.com/tensorflow/tensorflow/blob/e832dcc31c3c68b86c9c4ab5ebbb78cfdd28dcee/tensorflow/python/keras/metrics_test.py#L1977 Can we keep consistency with tf-core/keras?